web3/process.go

```diff
@@ -119,90 +138,137 @@ func (c *Contracts) StateRoot(processID []byte) (*types.BigInt, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
 	defer cancel()
-	if blobsSidecar == nil { // Regular transaction if no blobs are provided
-		autOpts, err := c.authTransactOpts()
-		if err != nil {
-			return nil, fmt.Errorf("failed to create transact options: %w", err)
-		}
-		autOpts.Context = ctx
-		tx, err := c.processes.SubmitStateTransition(autOpts, pid, proof, inputs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to submit state transition: %w", err)
-		}
-		hash := tx.Hash()
-		return &hash, nil
-	}
-
+	// Prepare the ABI for packing the data
 	processABI, err := c.ProcessRegistryABI()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get process registry ABI: %w", err)
+		return nil, nil, fmt.Errorf("failed to get process registry ABI: %w", err)
 	}
-	tx, err := c.NewEIP4844Transaction(
-		ctx,
-		c.ContractsAddresses.ProcessRegistry,
-		processABI,
-		"submitStateTransition",
-		[]any{pid, proof, inputs},
-		blobsSidecar,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create EIP-4844 transaction: %w", err)
+	// Use transaction manager for automatic nonce management
+	var sentTx *gtypes.Transaction
+	txID, txHash, err := c.txManager.SendTx(ctx, func(nonce uint64) (*gtypes.Transaction, error) {
+		internalCtx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
+		defer cancel()
+
+		var tx *gtypes.Transaction
+		if blobsSidecar == nil {
+			// Regular transaction
+			data, dataErr := processABI.Pack("submitStateTransition", pid, proof, inputs)
+			if dataErr != nil {
+				return nil, fmt.Errorf("failed to pack data: %w", dataErr)
+			}
+
+			tx, err = c.txManager.BuildDynamicFeeTx(internalCtx, c.ContractsAddresses.ProcessRegistry, data, nonce)
+		} else {
+			// Blob transaction
+			tx, err = c.NewEIP4844TransactionWithNonce(
+				internalCtx,
+				c.ContractsAddresses.ProcessRegistry,
+				processABI,
+				"submitStateTransition",
+				[]any{pid, proof, inputs},
+				blobsSidecar,
+				nonce,
+			)
+		}
+		if err == nil {
+			sentTx = tx // Store the transaction for sidecar tracking
+		}
+		return tx, err
+	})
+	// If blob transaction sent successfully, store sidecar for recovery
+	if err == nil && blobsSidecar != nil && sentTx != nil {
+		if err := c.txManager.TrackBlobTxWithSidecar(sentTx, blobsSidecar); err != nil {
+			log.Warnw("failed to track blob sidecar for recovery",
+				"error", err,
+				"hash", txHash.Hex(),
+				"txID", fmt.Sprintf("%x", txID))
+		} else {
+			log.Infow("blob sidecar tracked for stuck transaction recovery",
+				"hash", txHash.Hex(),
+				"txID", fmt.Sprintf("%x", txID),
+				"blobCount", len(blobsSidecar.Blobs))
+		}
 	}
+	log.Infow("state transition submitted",
+		"processID", fmt.Sprintf("%x", processID))
+	return txID, txHash, err
+}
 
-	if err := c.cli.SendTransaction(ctx, tx); err != nil {
-		return nil, fmt.Errorf("failed to submit state transition (EIP-4844 Tx): %w", err)
+func (c *Contracts) SetProcessTransitionAndWait(
```

Comment:
We do we need SetProcessTransitionAndWait() and SetProcesstransition() ? Where is the use case that do not require waiting? Please, try to simplify package APIs. Just export those necessary and try not to repeat functionality in two different methods if It's not actually necessary.

Reply:
I tried to keep the old API available and just extend it, but ok, i will make them private or remove they.

Reply:
Everything not needed, just remove it.